data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraSourceConfigTest.java

```diff
@@ -0,0 +1,103 @@
+package org.opensearch.dataprepper.plugins.source.jira;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.BASIC;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.OAUTH2;
+
+public class JiraSourceConfigTest {
+    private JiraSourceConfig jiraSourceConfig;
+
+    private String accessToken = "access token test";
+    private String refreshToken = "refresh token test";
+    private String clientId = "client id test";
+    private String clientSecret = "client secret test";
+
+
+    private JiraSourceConfig createJiraSourceConfig(String authtype, boolean hasToken) throws JsonProcessingException {
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("account_url", "https://example.atlassian.net");
+
+        Map<String, String> connectorCredentialMap = new HashMap<>();
+        connectorCredentialMap.put("auth_type", authtype);
+        if (hasToken) {
+            connectorCredentialMap.put("access_token", accessToken);
+            connectorCredentialMap.put("refresh_token", refreshToken);
+        } else {
+            connectorCredentialMap.put("refresh_token", "");
+        }
+        connectorCredentialMap.put("client_id", clientId);
+        connectorCredentialMap.put("client_secret", clientSecret);
+
+        configMap.put("connector_credentials", connectorCredentialMap);
+
+        List<String> projectList = Arrays.asList("project1", "project2");
+        configMap.put("project", projectList);
+
+        List<String> issueTypeList = Arrays.asList("issue type 1", "issue type 2");
+        configMap.put("issue_type", issueTypeList);
+
+        List<String> inclusionPatternList = Arrays.asList("pattern 1", "pattern 2");
+        configMap.put("inclusion_patterns", inclusionPatternList);
+
+        List<String> exclusionPatternList = Arrays.asList("pattern 3", "pattern 4");
+        configMap.put("exclusion_patterns", exclusionPatternList);
+
+        List<String> statusList = Arrays.asList("status 1", "status 2");
+        configMap.put("status", statusList);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonConfig = objectMapper.writeValueAsString(configMap);
+        JiraSourceConfig config = objectMapper.readValue(jsonConfig, JiraSourceConfig.class);
+        return config;
+    }
+
+    @Test
+    void testGetters() throws JsonProcessingException {
+        jiraSourceConfig = createJiraSourceConfig(BASIC, false);
+        assertNotNull(jiraSourceConfig.getInclusionPatterns());
```

Comment:
Ideally, this should check if the return values match with the expected values. Checking for non-null is not enough

Reply:
changing to assertEquals